src/main/kotlin/org/jetbrains/teamcity/rest/api.kt

```diff
@@ -181,3 +183,18 @@ interface TriggeredInfo {
     val user: User?
     val build: Build?
 }
+
+interface ArtifactDependency {
+    val dependsOnBuildConfiguration: BuildConfiguration
+    val branch : String?
+    val artifactRules: List<ArtifactRule>
+    val cleanDestinationDirectory: Boolean
+}
+
+interface ArtifactRule {
+    val rule: String
+    val include: Boolean
+    val sourcePath: String
```

Comment:
According to TeamCity [docs](https://confluence.jetbrains.com/display/TCD10/Artifact+Dependencies) the paths may include wildcards. I think it's worth to mention that in javadoc for there properties.